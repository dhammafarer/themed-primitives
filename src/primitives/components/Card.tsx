import * as React from "react";
import { styled, css, Scale } from "../../theme";
import { Flex, flex, FlexProps } from "./Flex";
import {
  getOpacity,
  getBoxShadow,
  getBorder,
  getBorderRadius,
  getBorderColor,
} from "../getters";

interface CardProps extends FlexProps {
  opacity?: number;
  shadow?: Scale;
  radius?: Scale;
  b?: Scale;
  br?: Scale;
  bl?: Scale;
  bt?: Scale;
  bb?: Scale;
  bx?: Scale;
  by?: Scale;
  borderColor?: string;
}

const card = css<CardProps>`
  overflow: hidden;
  flex-direction: column;
  ${flex}
  ${props => css`
    ${getOpacity(props)}
    ${getBoxShadow(props)}
    ${getBorder(props)}
    ${getBorderColor(props)}
    ${getBorderRadius(props)}
  `}
`;

const Card = styled(
  ({ opacity, boxShadow, b, borderColor, radius, ...rest }) => (
    <Flex {...rest} />
  )
)<CardProps & React.HTMLProps<HTMLDivElement>>`
  ${card}
`;

export { Card, CardProps };
