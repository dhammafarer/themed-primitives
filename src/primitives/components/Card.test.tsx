import * as React from "react";
import * as renderer from "react-test-renderer";

import { Card } from "./Card";

it("should render a div", () => {
  const tree = renderer.create(<Card />).toJSON();

  expect(tree).toMatchSnapshot();
});
