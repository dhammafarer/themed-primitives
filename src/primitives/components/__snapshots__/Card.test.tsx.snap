// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a div 1`] = `
<div
  className="sc-htpNat jqpHFZ sc-bwzfXH cZTpwT sc-bdVaJa jolpEI"
/>
`;
